### PR TITLE
Remove Cube-specific DistanceManager#addPlayer and removePlayer. Closes #51

### LINKED ISSUE #51: Remove MixinDistanceManager.addCubePlayer and use addPlayer directly since it uses section pos
Would that work?

### PR DESCRIPTION
**Pull request by pufmat**

### CODE REVIEW

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/ticket/MixinDistanceManager.java [hunk 1]
```diff
@@ -213,18 +214,25 @@ public void updateCubeForced(CubePos pos, boolean add) {
...
-    @Override
-    public void addCubePlayer(CubePos cubePos, ServerPlayer player) {
-        long cubePosLong = cubePos.asLong();
+    @Inject(method = "addPlayer", at = @At("RETURN"))
+    public void addCubePlayer(SectionPos sectionPos, ServerPlayer player, CallbackInfo ci) {
+        long cubePosLong = CubePos.asLong(
+            Coords.sectionToCube(sectionPos.getX()),
+            Coords.sectionToCube(sectionPos.getY()),
+            Coords.sectionToCube(sectionPos.getZ())
+        );
```
Comment (Niko-sk2x) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/ticket/MixinDistanceManager.java:226]: these probably now need to check for non-CC world

**Comment by NotStirred** [comment]
This PR actually brings up something interesting. We shouldn't need chunk player tickets at all, they should be handled fully by `COLUMN` tickets.

I'm not sure if any mods depend on ticket type - though it would be weird if they did

**Comment by NotStirred** [comment]
Discussion on discord [here](https://discord.com/channels/316679487955927050/317206370359443458/939699465290416128).
We should replace the vanilla implementation instead of keeping both. Perhaps a `cancellable` `Inject` at `HEAD`?

**Comment by NotStirred** [comment]
After looking into it a bit more, it's probably best to not do this, it's a bit of a rabbit hole sadly

**Pull request merged**